packages/validator/src/util/clock.ts

```diff
@@ -70,15 +70,22 @@ export class Clock implements IClock {
     this.fns.push({timeItem: TimeItem.Epoch, fn});
   }
 
-  /** Milliseconds from now to a specific slot */
-  msToSlot(slot: Slot): number {
-    const timeAt = this.genesisTime + this.config.SECONDS_PER_SLOT * slot;
-    return timeAt * 1000 - Date.now();
+  /** Milliseconds from now to basis point into a specific slot */
+  msToSlot(slot: Slot, basisPoints = 0): number {
+    const timeAt =
+      (this.genesisTime + this.config.SECONDS_PER_SLOT * slot) * 1000 +
+      getSlotComponentDuration(this.config, basisPoints);
+    return timeAt - Date.now();
   }
 
-  /** Seconds elapsed from a specific slot to now */
-  secFromSlot(slot: Slot): number {
-    return Date.now() / 1000 - (this.genesisTime + this.config.SECONDS_PER_SLOT * slot);
+  /** Seconds elapsed from `basisPoints` into a specific slot to now */
+  secFromSlot(slot: Slot, basisPoints = 0): number {
```

Comment:
Would prefer to either rename this function to properly describe the use of second parameter or have a new function. e.g. `secFromSlotBasePoint` or `secFromSlotComponent`